omnigan/trainer.py

```diff
@@ -926,11 +1059,34 @@ def save(self):
 
     def resume(self):
         # load_path = self.get_latest_ckpt()
-        load_path = Path(self.opts.output_path) / Path("checkpoints/latest_ckpt.pth")
-        checkpoint = torch.load(load_path)
-        print(f"Resuming model from {load_path}")
+        if "m" in self.opts.tasks and "p" in self.opts.tasks:
+            m_path = self.opts.load_paths.m
+            p_path = self.opts.load_paths.p
+
+            if m_path == "none":
+                m_path = self.opts.output_path
+            if p_path == "none":
+                p_path = self.opts.output_path
+
+            # Merge the dicts
+            m_ckpt_path = Path(m_path) / Path("checkpoints/latest_ckpt.pth")
+            p_ckpt_path = Path(p_path) / Path("checkpoints/latest_ckpt.pth")
+
+            m_checkpoint = torch.load(m_ckpt_path)
+            p_checkpoint = torch.load(p_ckpt_path)
+
+            checkpoint = merge(m_checkpoint, p_checkpoint)
+            print(f"Resuming model from {m_ckpt_path} and {p_ckpt_path}")
+        else:
+            load_path = Path(self.opts.output_path) / Path(
+                "checkpoints/latest_ckpt.pth"
+            )
+            checkpoint = torch.load(load_path)
+            print(f"Resuming model from {load_path}")
+
         self.G.load_state_dict(checkpoint["G"])
-        self.g_opt.load_state_dict(checkpoint["g_opt"])
+        if not ("m" in self.opts.tasks and "p" in self.opts.tasks):
```

Comment:
basically loading if not training both. why?

Reply:
I'm not sure what to do with the optimizer otherwise. Which optimizer should we load if we have two different experiments? I thought the safe bet would be to start over